binlog_streamer.go

```diff
@@ -161,12 +162,18 @@ func (s *BinlogStreamer) Run() {
 		func() {
 			ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
 			defer cancel()
+
+			getEventStart := time.Now()
 			ev, err = s.binlogStreamer.GetEvent(ctx)
 			if err == context.DeadlineExceeded {
 				timedOut = true
 			} else if err != nil {
 				s.ErrorHandler.Fatal("binlog_streamer", err)
 			}
+
+			metrics.Timer("BinlogStreamer.GetEvent", time.Since(getEventStart), []MetricTag{
+				MetricTag{"DeadlineExceeded", strconv.FormatBool(err == context.DeadlineExceeded)},
+			}, 1.0)
```

Comment:
In the past case of the lost connection, it would have been nice to see it not getting events but being in the loop. I think we can work on the metrics sink full by reducing the volume of total events. I'll try to aggregate how many events a typical events emit and try to balance the volume a bit with rates.